### ISSUE TITLE
"function hideSplash" is not calling plymouth hide-splash when "console=tty0 console=ttyS0" is added to kernel options

### ISSUE THREAD

**Issue by ipeoshir** [openSUSE/kiwi#623]
Description:
plymouth splash is still active while the dialog to select the disk to install OEM images are on screen and any keyboard input makes a "loading" screen to appear when keys are typed. We have to manually open the console (Alt+F2) and call "plymouth hide-splash" to have this "loading" screen disapper.

Line "$console_count -eq 1" from kiwi's "/include" script is not handling the scenario when tty0 and ttyS0 is used to multiplex the consoles on kernel options and "plymouth hide-splash" is never called.

Setup:
File "/srv/tftpboot/pxelinux.cfg/default" on PXE server will need to have the following kernel options:

> LABEL KIWI-Boot
>     kernel boot/linux
>     append initrd=boot/initrd pxe=1 kiwidebug=1 console=tty0 console=ttyS0,57600n8
>     IPAPPEND 1

Then we create an OEM pxe image installer with kiwi like this:

> kiwi --verbose 3 --yes --build /root/suse-SLE12-build --add-profile oemFlavour --destdir /root/suse-SLE12-oemboot

The images need to be transfered to the pxe server to boot. To test it on qemu, we need multiple disks to prompt the dialog with the selected option for the installation:

> virt-install --connect qemu:///system --name linux-01 --ram 4096 --os-type=linux --disk path=/tmp/disk1.raw,bus=scsi --disk path=/tmp/disk2.raw,bus=scsi --disk path=/tmp/disk3.raw,bus=scsi --pxe --vcpus=1 --network network=default --boot hd,network,menu=on --video=vga

Also, there is a plymouth bug interfering the installation (it's consuming keys on serial output making the screen to become distorted) and this can be verified if the package used to create initrd has this patch applied:

- https://build.opensuse.org/package/view_file/SUSE:SLE-12-SP2:GA/plymouth/plymouth-filter-escape-codes.patch?expand=1

Reference:
- https://groups.google.com/d/msg/kiwi-images/3LN2wlZ-ims/ZYhJjP1uBgAJ

**Comment by schaefi** [comment]
Thanks for diving so deeply into this issue, I'll have a look regarding the kiwi part

**Comment by davidcassany** [comment]
Yes @iposhir thanks for this elaborated report. 

May I ask which KIWI version are you using? I am asking because in #601 we added a patch to ensure `hideSplash` method was calling `plymouth hide-splash` regardless the console status. The line `$console_count -eq 1` should not be there anymore. Just checking in which sle version is the patch included.

**Comment by ipeoshir** [comment]
SLES12Sp2 one:
> kiwi-7.03.96-48.5.x86_64.rpm

**Comment by ipeoshir** [comment]
Actually, I was debuging all those from SLES12 versions to see if that would help me with the screen problem:

drwxr-xr-x 2 root root 4096 Jan 12 12:44 5.06.165-1.7
drwxr-xr-x 2 root root 4096 Jan 12 12:49 5.06.182-15.1
drwxr-xr-x 2 root root 4096 Jan 12 13:36 5.06.195-17.11.7
drwxr-xr-x 2 root root 4096 Jan 16 07:33 7.02.77-1.1
drwxr-xr-x 2 root root 4096 Jan 13 10:46 7.03.96-48.5

**Comment by schaefi** [comment]
I have checked the SP2 (internal) update channel. The following package is on the way to the public channel if not already there:

   SUSE/Updates/SLE-SERVER/12-SP2/x86_64/update/x86_64/kiwi-7.03.108-60.1.x86_64.rpm

This one contains the fix. If you would like to give it a try you can also use the version from the
development project.

* http://download.opensuse.org/repositories/Virtualization:/Appliances/SLE_12_SP2/

I'm afraid I can't tell when the SP2 update channel will be synced with the internal tree

**Comment by ipeoshir** [comment]
Hmmm... I updated my kiwi's version here and now the boot image halts:

![kiwi-halts](https://cloud.githubusercontent.com/assets/16229081/22026168/2927d940-dcb7-11e6-8d4b-9d7b38df82e7.png)

**Comment by ipeoshir** [comment]
Those versions were installed from the update channel:

# rpm -qa | grep kiwi
kiwi-7.03.108-60.1.x86_64
kiwi-desc-netboot-7.03.108-60.1.x86_64
kiwi-tools-7.03.108-60.1.x86_64
kiwi-desc-vmxboot-7.03.108-60.1.x86_64
kiwi-desc-isoboot-7.03.108-60.1.x86_64
kiwi-desc-oemboot-7.03.108-60.1.x86_64
kiwi-templates-7.03.108-60.1.x86_64

**Comment by ipeoshir** [comment]
I thought it could be a kernel issue (SLES12SP0 build too old for SLES12SP2 updated by kiwi), but no... See the same with kernel-default-4.4.21-69.1.x86_64.rpm:

![kiwi-halts-new-kernel](https://cloud.githubusercontent.com/assets/16229081/22027542/547960ba-dcbb-11e6-8270-0a08b955e30f.png)

**Comment by davidcassany** [comment]
mmmm it needs further investigation then, could you share with us `<type>` section you are using in your kiwi description file? I'll try to build an image similar to yours to reproduce it.

When it breaks have you been able to enter to a simple shell pressing Enter? if so `/var/log/boot.kiwi` might help to understand where and why it failed.

**Comment by ipeoshir** [comment]
$ kiwi --verbose 3 --yes --build /root/suse-SLE12-build --add-profile oemFlavour --destdir /root/suse-SLE12-oemboot

>     <preferences profiles="oemFlavour">
>         <type image="oem" filesystem="ext3" boot="oemboot/suse-SLES12" installpxe="true" bootloader="grub2" firmware="uefi" hybrid="true" kernelcmdline="splash" primary="true">
>             <oemconfig>
>                 <oem-systemsize>2048</oem-systemsize>
>                 <oem-swap>true</oem-swap>
>             </oemconfig>
>             <machine memory="512" guestOS="suse" HWversion="4">
>                 <vmdisk id="0" controller="lsilogic"/>
>                 <vmnic driver="e1000" interface="0" mode="bridged"/>
>             </machine>
>         </type>
>     </preferences>

Looks like it could not get an IP: "Can't get DHCP reply on any interface !". I could not connect to tftp file to put the log file to paste it here, but the exception was the one above...

**Comment by ipeoshir** [comment]
OK, I reverted it back from "7.03.108-60.1" to "7.03.96-48.5", rebuilt my images and the exception was not raised. It seems it is some sort of regression issue between those 2 kiwi builds.

**Comment by davidcassany** [comment]
@ipeoshir I managed to reproduce a broken dialog in my local environment, I am using kiwi 7.04.21, however I could not see the exception you mention above that you saw in 7.03.108 version. I am reopening the issue as it is not solved yet. 

Reopening the issue. Tomorrow I will keep on debugging it.

**Comment by ipeoshir** [comment]
Thanks, David.

The broken dialog is most likely related to plymouth, not kiwi. We are investigating it as well. The network changes that were added post 7.03.96 though might be a concern if it starts breaking OEM PXE installations via network.

**Comment by davidcassany** [comment]
@ipeoshir you are right it concerns us if it breaks OEM PXE installs. I know there is a plymouth issue which might not be related to KIWI, however I found something weird while reviewing kiwi logs and I am trying to figure out if there isn't also a problem with the KIWI boot code when the serial console is enabled.

**Comment by davidcassany** [comment]
@ipeoshir I finally managed to deploy oem images via PXE without issues and with all the dialogs. Apparently in my first tests I was not connecting properly to the serial console in my libvirt environment. I managed to use it regardless the plymouth was enabled or not, so even there might be an issue with plymouth, I'd say is not effecting us.

This is the append line I used:
`append initrd=boot/initrd vga=normal console=tty0 console=ttyS0,57600n8 kiwiserver=192.168.122.50 pxe=1 splash`

Note that dialog is only sent to the serial console with this configuration.
